web/src/main/java/edu/softserve/zoo/DtoMapper.java

```diff
@@ -0,0 +1,128 @@
+package edu.softserve.zoo;
+
+import edu.softserve.zoo.annotation.Dto;
+import edu.softserve.zoo.annotation.IrrespectiveDto;
+import edu.softserve.zoo.dto.BaseDto;
+import edu.softserve.zoo.model.BaseEntity;
+import org.apache.commons.collections4.BidiMap;
+import org.apache.commons.collections4.bidimap.DualHashBidiMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.core.type.filter.TypeFilter;
+import org.springframework.util.ClassUtils;
+
+import javax.annotation.PostConstruct;
+import java.lang.annotation.Annotation;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+
+/**
+ * Class that manages relationships between Entities and their DTO.
+ */
+public class DtoMapper {
```

Comment:
@skyfenko 
no, actually, it isn't.
It is declared as bean in xml config.
But alternatively, we can mark it as @Component.
Have I understood you correctly?